### ISSUE TITLE
Fix version of documentation depdencies

### ISSUE THREAD

**Issue by AdamRJensen** [jranalli/solarspatialtools#24]
The versions of the documentation dependencies should be fixed to avoid the documentation breaking when new releases of those packages come out.

**Comment by jranalli** [comment]
All libraries are pinned with maximum versions in #13

**Comment by AdamRJensen** [comment]
Hi @jranalli 

It was actually these packages that I was recommending to be fixed: 
https://github.com/jranalli/solarspatialtools/blob/806d047ab847851c86466979108bcac6afc4ef7c/setup.cfg#L38-L43

I personally would not pin the upper versions of the other packages, as they will in most cases work for later versions of those packages.

**Comment by jranalli** [comment]
Whoops, I misunderstood. I can add those in. The other packages, one of the other reviewers suggested it as a concern, because recent numpy (and maybe pandas too) updates wreaked some havoc recently. I'll try to actually evaluate which of them are a problem at the moment and try to be more judicious about it.

**Comment by AdamRJensen** [comment]
Could you make it compatible with numpy version 2 and set a lower limit on numpy?

**Comment by jranalli** [comment]
Let me look at it. I believe the issue was that the latest pandas at the time wasn't numpy 2 compatible?

**Comment by jranalli** [comment]
Yep, the issue appears to be that pandas still is using a numpy<2 dependency. So there's some environment mess there. Let me see if my tests run with that as the only pin.

**Comment by jranalli** [comment]
Alright, I'm sorry, I was misinterpreting something. There is some weird incompatibility between numpy and pandas HD5 reading that is OS and python version dependent. See for example [this run that fails with numpy 2.0](https://github.com/jranalli/solarspatialtools/actions/runs/10858057240/job/30135513353) and [this run that passes with numpy 1.X](https://github.com/jranalli/solarspatialtools/actions/runs/10864035038/job/30148860694). 

My development is done on a Windows computer and so I think I was running into that issue with my own environment and had some confusion. And then I got faked out by trying to look into what dependencies pandas was specifying, which is sometimes [pinned below numpy 2](https://github.com/pandas-dev/pandas/blob/main/environment.yml), and there are plenty of web results for that error message talking about incompatibility between pandas and numpy 2, possibly from right around when numpy 2 was released.

I am not adept enough with the environment management to figure out exactly what those bugs are that are affecting other versions and whether they're caused by a poor installation of pandas/numpy or something else. Now that I can see what's going on though, it appears that everything here is numpy 2 compatible as long as I limit my tests to recent versions of python (>3.10). So I think this one is closed out.
